### ISSUE TITLE
Provide a TypeScript type for the `.eslintrc.js` configuration file format

### ISSUE THREAD

**Issue by bmish** [eslint/eslint#14586]
**The problem you want to solve.**

I have various Node/TypeScript scripts that read the contents of `.eslintrc.js` in my packages/monorepos to check that things are configured consistently and correctly. Surely, there are various use cases for scripts needing to read, check, and even modify lint configuration values.

For TypeScript code, it is of course much easier and safer working with typed data, and so I've been interested to find a complete TypeScript type for `.eslintrc.js`.

**Your take on the correct solution to problem.**

There are a few options for where this type data could be stored and provided from:

* ESLint (my preference)
  * Bundling the types here would guarantee that a consumer of ESLint would receive the exact type data for the version they're using
  * Anyone using ESLint may be interested in this, regardless of whether or not they are actually linting TypeScript code
* [@typescript-eslint/](https://github.com/typescript-eslint/typescript-eslint)
  * Has lots of existing TypeScript related code and functionality 
* [type-fest](https://github.com/sindresorhus/type-fest#miscellaneous)
  * This third-party library already provides related types for [package.json](https://github.com/sindresorhus/type-fest/blob/main/source/package-json.d.ts) and [tsconfig.json](https://github.com/sindresorhus/type-fest/blob/main/source/tsconfig-json.d.ts)

I could then do something like this when I want to work with the ESLint configuration data in my script:

```ts
import type { ESLintRC } from 'eslint';

const config: ESLintRC = require('.eslintrc.js');

// TypeScript would actually error here because it knows that `config.plugins` can be undefined
config.plugins.forEach(plugin => {
  // do something with each plugin
});
```

Very rough start of what the type would look like:

```ts

export declare type ESLintRC = {
  root?: boolean;
  parser?: string;
  parserOptions?: { ... };
  plugins?: string[];
  extends?: string[];
  env?: { es6?: boolean, node?: boolean, ... };
  rules?: { ... };
  overrides?: { ... }[];
};
```

**Are you willing to submit a pull request to implement this change?**

Maybe.

**Comment by nzakas** [comment]
Thanks for the suggestion. We are purposely staying away from using TypeScript in this project as we are JavaScript-focused. As we are in the process of moving away from eslintrc, we also don’t want to be making any further changes related to it.

Related: https://github.com/eslint/eslint/issues/14249
